src/porting.cpp

```diff
@@ -697,6 +699,20 @@ int mt_snprintf(char *buf, const size_t buf_size, const char *fmt, ...)
 	return c;
 }
 
+bool openURL(std::string url)
+{
+	if (!std::regex_match(url, std::regex("^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$") )) {
+		errorstream << "Invalid url: " << url << std::endl;
+		return false;
+	}
+
+#ifdef _WIN32
+	return system((std::string("open ") + url).c_str()) == 0;
+#else
+	return system((std::string("xdg-open ") + url).c_str()) == 0;
```

Comment:
I'm really not happy with this method of opening URLs, but I couldn't find a better way

Reply:
Did you try this on android? IIRC you will need to use JNI/java for this I havn't tested it. Maybe `xdg-open` works there, too.

Reply:
yeah, this doesn't support Android yet. I need to find out how to call Java functions from this part of the code, so I can then use intents to open the browser